pkg/cloudprovider/azure/object_store.go

```diff
@@ -37,10 +41,57 @@ func NewObjectStore(logger logrus.FieldLogger) cloudprovider.ObjectStore {
 	return &objectStore{log: logger}
 }
 
+func getStorageAccountsClient(envVars map[string]string) (*storagemgmt.AccountsClient, error) {
+	spt, err := helpers.NewServicePrincipalTokenFromCredentials(envVars, azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating new service principal token")
+	}
+
+	accountsClient := storagemgmt.NewAccountsClient(envVars[azureSubscriptionIDKey])
+	accountsClient.Authorizer = autorest.NewBearerAuthorizer(spt)
+
+	return &accountsClient, nil
+}
+
+func getStorageAccountKey(client *storagemgmt.AccountsClient, resourceGroup, storageAccount string) (string, error) {
+	res, err := client.ListKeys(resourceGroup, storageAccount)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	if res.Keys == nil || len(*res.Keys) == 0 {
+		return "", errors.New("No storage keys found")
+	}
+
+	var storageKey string
+
+	for _, key := range *res.Keys {
+		// uppercase both strings for comparison because the ListKeys call returns e.g. "FULL" but
```

Comment:
Do we need to update the docs to make it 100% clear that we require a storage account key with "full" permissions? Could we get by with fewer?

Reply:
Hmm I see that now, too. I was looking at your link on Azure key management practices, and it mentioned different storage account RBAC permissions, including owner, contributor, reader, etc.